### ISSUE TITLE
Sentinel client does not listen to failover events

### ISSUE THREAD

**Issue by ederuiter** [redis-rs/redis-rs#1438]
Currently the redis-rs sentinel client only uses the sentinel connection to determine the correct redis server when initiating a new connection. It currently lacks support for listening/reacting to the pub sub messages of the sentinel (https://redis.io/docs/latest/operate/oss_and_stack/management/sentinel/#pubsub-messages) 

This leads to issues with active connections not being switched over to a the new master.

Looking at the [golang client](https://github.com/redis/go-redis/blob/e63669e1706936ac794277340c51a51c5facca70/sentinel.go#L723); it listens for `+switch-master` and `+replica-reconf-done`

I can help with implementation; however my current rust experience is very limited; so I would need some pointers on how to approach this.

**Comment by nihohit** [comment]
> it will actively close all current connections to the previous master.

I don't think this can work - a connection is independent, and can't be closed externally. We could insert some backdoor that  kills the connection, but that would require shared state, and that's not simple to implement in Rust. You can look in `connection-manager.rs` to see how complex it is to keep a single connection and replace it when it disconnects.

But what if we don't kill the old connection? Maybe just have a channel sending the most recent connection, when we get a new one? that will let the user manually kill the old connection, if they want.

**Comment by ederuiter** [comment]
What about re-using that same mechanism? 
If you already can replace a connection when it disconnects; how about "just" reconnecting to the new master/replica?

Basically special-casing the reconnect for a connection initiated by a sentinelclient

**Comment by nihohit** [comment]
well, it would require using the connection manager in the sentinel client. ATM that's not supported.

**Comment by SamPeng87** [comment]
Expect the Sentinel client to support connection manager.

**Comment by nihohit** [comment]
@SamPeng87 you're welcome to contribute such a functionality. Although, maybe it would be simpler to receive a `redis::Client` from the sentinel, and let the user create their own connection from the client.
